xbmc/addons/interfaces/GUI/ListItem.cpp

```diff
@@ -213,34 +213,14 @@ char* Interface_GUIListItem::get_icon_image(void* kodiBase, void* handle)
 
   char* ret;
   Interface_GUIGeneral::lock();
-  ret = strdup(item->get()->GetIconImage().c_str());
+  ret = strdup(item->get()->GetArt("icon").c_str());
   Interface_GUIGeneral::unlock();
   return ret;
 }
 
 void Interface_GUIListItem::set_icon_image(void* kodiBase, void* handle, const char *image)
 {
-  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
-  CFileItemPtr* item = static_cast<CFileItemPtr*>(handle);
-  if (!addon || !item || !image)
-  {
-    CLog::Log(LOGERROR,
-              "Interface_GUIListItem::%s - invalid handler data (kodiBase='%p', handle='%p', "
-              "image='%p') on addon '%s'",
-              __FUNCTION__, kodiBase, handle, image, addon ? addon->ID().c_str() : "unknown");
-    return;
-  }
-
-  if (item->get() == nullptr)
-  {
-    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - empty list item called on addon '%s'",
-              __FUNCTION__, addon->ID().c_str());
-    return;
-  }
-
-  Interface_GUIGeneral::lock();
-  item->get()->SetIconImage(image);
-  Interface_GUIGeneral::unlock();
+  CLog::Log(LOGWARNING, "Usage of setIconImage results in NOP and will soon be removed.");
 }
 
```

Comment:
`get_icon_image` and `set_icon_image` can be removed complete, no more needed if not called on headers.